terraform/ec2/amis.tf

```diff
@@ -65,26 +65,185 @@ EOF
+      os_family = "ubuntu"
       ami_search_pattern = "ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-*"
       ami_owner = "099720109477"
-      family = "ubuntu"
+      family = "debian"
+      arch = "amd64"
+      login_user = "ubuntu"
+    }
+    ubuntu18 = {
+      os_family = "ubuntu"
+      ami_search_pattern = "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server*"
+      ami_owner = "099720109477"
+      family = "debian"
       arch = "amd64"
+      login_user = "ubuntu"
+    }
+    debian10 = {
+      os_family = "debian"
+      ami_search_pattern = "debian-10-amd64*"
+      ami_owner = "679593333241"
+      family = "debian"
+      arch = "amd64"
+      login_user = "admin"
+    }
+    debian9 = {
+      os_family = "debian"
+      ami_search_pattern = "debian-stretch-hvm-x86_64-gp2*"
+      ami_owner = "679593333241"
+      family = "debian"
+      arch = "amd64"
+      login_user = "admin"
     }
     amazonlinux2 = {
+      os_family = "amazon_linux"
       ami_search_pattern = "amzn2-ami-hvm*"
       ami_owner = "amazon"
       family = "amazon_linux"
       arch = "amd64"
+      login_user = "ec2-user"
     }
     windows2019 = {
+      os_family = "windows"
       ami_search_pattern = "Windows_Server-2019-English-Full-Base-*"
       ami_owner = "amazon"
       family = "windows"
       arch = "amd64"
+      login_user = "Administrator"
+    }
+    amazonlinux = {
+      os_family = "amazon_linux"
+      ami_search_pattern = "amzn-ami-hvm*"
+      ami_owner = "amazon"
+      family = "amazon_linux"
+      arch = "amd64"
+      login_user = "ec2-user"
+    }
+    suse15 = {
+      os_family = "suse"
+      ami_search_pattern = "suse-sles-15*"
+      family = "linux"
+      login_user = "ec2-user"
+      arch = "amd64"
+      ami_owner = "amazon"
+    }
+    suse12 = {
+      os_family = "suse"
+      ami_search_pattern = "suse-sles-12*"
+      family = "linux"
+      login_user = "ec2-user"
+      arch = "amd64"
+      ami_owner = "amazon"
+    }
+    redhat8 = {
+      os_family = "redhat"
+      ami_search_pattern = "RHEL-8.0.0_HVM*"
+      family = "linux"
+      ami_owner = "309956199498"
+      arch = "amd64"
     }
+    redhat7 = {
+      os_family = "redhat"
+      ami_search_pattern = "RHEL-7.7_HVM*"
+      family = "linux"
+      ami_owner = "309956199498"
+      arch = "amd64"
+    }
+    centos7 = {
+      login_user = "centos"
+      os_family = "centos"
+      ami_search_pattern = "CentOS Linux 7 x86_64*"
+      family = "linux"
+      ami_owner = "679593333241"
+      arch = "amd64"
+    }
+    centos6 = {
+      login_user = "centos"
+      os_family = "centos"
+      ami_search_pattern = "CentOS Linux 6 x86_64 HVM EBS 2002*"
+      family = "linux"
+      ami_owner = "679593333241"
+      arch = "amd64"
+      user_data = <<EOF
+#! /bin/bash
+sudo iptables -I INPUT -p tcp -m tcp --dport 55680 -j ACCEPT
+sudo iptables -I INPUT -p tcp -m tcp --dport 55690 -j ACCEPT
+sudo service iptables save
```

Comment:
just curious. why CentOS need update iptables but other OSs do not need?

Reply:
Yeah, centos6 blocks almost all the ports by default